app/controllers/budgets/investments_controller.rb

```diff
@@ -77,6 +77,26 @@ def suggest
       super
     end
 
+    def edit_image
+    end
+
+    def update_image
+      if @investment.update(investment_params)
+        redirect_to budget_investment_path(@investment.budget, @investment),
+                    notice: t("flash.actions.update_image.budget_investment")
+      else
+        render :edit_image
+      end
+    end
+
+    def remove_image
+      @investment.image = nil
```

Comment:
@bertocq i read the link you pasted and noticed that both solutions has the same effect, both removes image, but i prefer to use destroy method explictly over setting attachment object to nil. I think it will be easier to understand for other developers ... I'll try it locally and see if the image is really removed although also i try to make an spec to check that behavior.